internal/app/secretless/providers/kubernetes-secrets/provider.go

```diff
@@ -0,0 +1,115 @@
+package kubernetes_secrets
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/api/core/v1"
+)
+
+// Provider provides data values from Kubernetes Secrets.
+type Provider struct {
+	Name   string
+	Client *KubeClient
+}
+
+// ProviderFactory constructs a Provider. The API client is configured from
+// in-cluster environment variables and files.
+func ProviderFactory(options plugin_v1.ProviderOptions) plugin_v1.Provider {
+	var client *KubeClient
+	var err error
+	if client, err = NewKubeClient(); err != nil {
+		log.Panicf("ERROR: Could not create Kubernetes Secrets provider: %s", err)
+	}
+
+	provider := Provider{
+		Name:   options.Name,
+		Client: client,
+	}
+
+	return provider
+}
+
+// GetName returns the name of the provider
+func (p Provider) GetName() string {
+	return p.Name
+}
+
+// parseKubernetesSecretID returns the secret id and field name.
+func parseKubernetesSecretID(id string) (string, string) {
+	tokens := strings.SplitN(id, "#", 2)
```

Comment:
We should check if the token count returned is 2 and that both have something before proceeeding

Reply:
addressed